docs/features/deserializing-events.md

```diff
@@ -4,10 +4,35 @@ layout: default
 ---
 
 ## Deserializing Events
-We provide support for deserializing events to typed objects:
+
+We provide support for deserializing events to typed event objects where the custom event payload is available via the `.GetPayload()` method.
+The original event payload can also be accessed via the `.Data` property.
 
 ```csharp
-var eventGridMessage = EventGridParser.Parse<BlobCreated>(rawEvent);
+// Parse from your custom event implementation with the `.Parse<>` function.
+EventGridBatch<NewCarRegistered> eventGridBatch = EventGridParser.Parse<NewCarRegistered>(rawEvent);
+
+// The event data type will be wrapped inside an `EventGridEvent<>` instance.
+NewCarRegistered eventGridMessage = eventGridBatch.Events.First();
+
+// The original event payload can now be accessed.
+CarEventData typedEventPayload = eventGridMessage.GetPayload();
+object untypedEventPaylaod = eventGridMessage.Data;
+```
+
+Or from event data type objects where the event data payload is available via the `.GetPayload()` method.
```

Comment:
I would highlight here that this can be used to retrieve built-in types for Azure services to make it more explicit or am I wrong?

Reply:
Yes, that would be good.

Reply:
Ah, this is already done.

Reply:
Not sure what you are referring to, the sample or?

I would also provide insights in it in the text

Reply:
No, in the description text there's already mentioning about the Azure SDK event types and that we support this.

Reply:
Ah now I see it, sorry about that!